STM32/DC/USB_DEVICE/App/usbd_cdc_if.c

```diff
@@ -247,19 +247,13 @@ static int8_t CDC_Control_FS(uint8_t cmd, uint8_t* pbuf, uint16_t length)
 		pbuf[6] = LineCoding.datatype;
     break;
 
-    case CDC_SET_CONTROL_LINE_STATE:{
+    case CDC_SET_CONTROL_LINE_STATE: {
         USBD_SetupReqTypedef * req = (USBD_SetupReqTypedef *)pbuf;
-        if(req->wValue &0x0001 != 0){
-      	  isComPortOpen = true;
-        } else {
-          isComPortOpen = false;
-        }
+        isComPortOpen = ((req->wValue & 0x0001) != 0);

```

Comment:
Are we sure the only  states  (req->wValue is & 0x0001) can take are 0 and 1? If yes, then fine. If no, then we should check this as the isComPortOpen could be set wrongly.

Reply:
Yep I am sure. (req->wValue & 0x0001) is logic bit operator, i.e. only the last bit will remain.